modules/ding_ipe_filter/ding_ipe_filter.module

```diff
@@ -0,0 +1,146 @@
+<?php
+/**
+ * @file
+ * Provide IPE Filter menu functionality.
+ */
+
+include_once 'ding_ipe_filter.features.inc';
+
+
+/**
+ * Implements hook_menu().
+ */
+function ding_ipe_filter_menu() {
+  $items['admin/config/user-interface/ipe-filter'] = array(
+    'title' => 'IPE pane filter settings',
+    'description' => 'Configure IPE filter default options.',
+    'weight' => 1,
+    'page callback' => 'drupal_get_form',
+    'page arguments' => array('ding_ipe_filter_default_settings_form'),
+    'access arguments' => array('access administration pages'),
+    'file' => 'ding_ipe_filter.admin.inc',
+  );
+  $items['admin/config/user-interface/ipe-filter/default'] = array(
+    'title' => 'Settings',
+    'description' => 'Default settings for IPE filter',
+    'type' => MENU_DEFAULT_LOCAL_TASK,
+    'weight' => 1,
+  );
+  $items['admin/config/user-interface/ipe-filter/panes'] = array(
+    'title' => 'IPE pane filter',
+    'description' => 'Choose what content should be displayed on the add content modal page.',
+    'page callback' => 'drupal_get_form',
+    'page arguments' => array('ding_ipe_filter_admin_panes_form'),
+    'access arguments' => array('access administration pages'),
+    'file' => 'ding_ipe_filter.admin.inc',
+    'type' => MENU_LOCAL_TASK,
+    'weight' => 2,
+  );
+
+  return $items;
+}
+
+/**
+ * Implements hook_theme().
+ */
+function ding_ipe_filter_theme() {
+  return array(
+    'ding_ipe_filter_table' => array(
+      'render element' => 'form',
+    ),
+  );
+}
+
+/**
+ * Implements theme_hook().
+ */
+function theme_ding_ipe_filter_table($variables) {
+  $form = $variables['form'];
+  $header = $form['#header'];
+
+  $content = array(
+    'header' => $header,
+    'rows' => array(),
+  );
+  foreach (element_children($form) as $row_index) {
+    if (isset($form[$row_index]['title'])) {
+      $content['rows'][] = array(
+        drupal_render($form[$row_index]['title']),
+        drupal_render($form[$row_index]['value']),
+      );
+    }
+    else {
+      $content['rows'][] = array(array(
+        'data' => drupal_render($form[$row_index]),
+          'colspan' => 2,
+          'header' => TRUE,
+        ),
+      );
+    }
+  }
+  $output = theme('table', $content);
+  $output .= drupal_render_children($form);
+  return $output;
+}
+
+/**
+ * Implements hook_theme_registry_alter().
+ */
+function ding_ipe_filter_theme_registry_alter(&$registry) {
+  global $user;
+  $ipe_roles = variable_get('ding_ipe_filter_roles', array());
+
+  if (!empty($user->roles)) {
+    $role_names = array_values($user->roles);
+    $user_roles = array_combine($role_names, $role_names);
+    $roles = array_intersect(array_keys($user_roles), $ipe_roles);
+  }
+  if (!empty($roles)) {
```

Comment:
Using the permission system to handle access to the filter would simplify this code as well.